lib/plugins/rest.js

```diff
@@ -36,6 +37,76 @@ exports.register = function (server, options, next) {
 			});
 		}
 	});
+
+	server.route({
+		method  : "GET",
+		path    : "/v1/instances/{instanceId}",
+
+		handler : function (request, reply) {
+
+			var instanceId = request.params.instanceId;
+
+			instances.getInstance({ id : instanceId })
+			.then (function (instance) {
+				if (!instance) {
+					throw Boom.notFound("No instances found for id " + instanceId);
+				}
+				reply(instance);
+			})
+			.catch (function (error) {
+				reply(Boom.wrap(error));
+			});
+		}
+	});
+
+	server.route({
+		method  : "GET",
+		path    : "/v1/instances",
+
+		handler : function (request, reply) {
+
+			var	query = {};
+			var bCheckPresenceofInValidParameter = false;
+			/*Create a JSON object of all paramaters which are present in the GET Request*/
+			for (var key in request.query) {
```

Comment:
I think this may benefit from using Lodash's [`_.pick` function](https://lodash.com/docs#pick).